app/src/main/java/org/openobservatory/ooniprobe/receiver/TestRunBroadRequestReceiver.java

```diff
@@ -0,0 +1,122 @@
+package org.openobservatory.ooniprobe.receiver;
+
+
+import android.content.BroadcastReceiver;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.ServiceConnection;
+import android.os.IBinder;
+
+import com.google.common.collect.Lists;
+import com.google.common.math.Stats;
+
+import org.openobservatory.ooniprobe.common.PreferenceManager;
+import org.openobservatory.ooniprobe.common.service.RunTestService;
+import org.openobservatory.ooniprobe.test.TestAsyncTask;
+import org.openobservatory.ooniprobe.test.suite.AbstractSuite;
+
+import java.util.List;
+
+public class TestRunBroadRequestReceiver extends BroadcastReceiver implements ServiceConnection {
```

Comment:
Please add some short documentation for this class.

Reply:
Also, I am bit confused by the name of this class. Shouldn't it be named `TestRunBroadcastReceiver`? Why is it named `Broad` and not `Broadcast`? Which `Request` is it receiving?